test/Microsoft.Azure.WebJobs.Host.UnitTests/AttributeClonerTests.cs

```diff
@@ -38,6 +38,34 @@ public Attr2(string resolvedProp2, string constantProp)
             public string ConstantProp { get; private set; }
         }
 
+        // Helper to easily generate a fixed binding contract. 
+        private static IReadOnlyDictionary<string, Type> GetBindingContract(params string[] names)
+        {
+            var d = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
+            foreach (var name in names)
+            {
+                d[name] = typeof(string);
+            }
+            return d;
+        }
+        private static IReadOnlyDictionary<string, Type> EmptyContract = new Dictionary<string, Type>();
+
+        // Enforce binding contracts statically. 
+        [Fact]
+        public void BindingContractMismatch()
+        {
+            Attr1 a1 = new Attr1 { Path = "{name}" };
+
+            try
+            {
+                var cloner = new AttributeCloner<Attr1>(a1, EmptyContract);
```

Comment:
I didn't use it because the helper won't let me verify the message contents.  

---
In reply to: [64429034](https://github.com/Azure/azure-webjobs-sdk/pull/731#discussion_r64429034) [](ancestors = 64429034)